TTHAnalysis/python/plotter/susy-sos/functionsSF.cc

```diff
@@ -4,770 +4,691 @@
-TGraphAsymmErrors* h_elSF_looseToTight_endcap_17 = (TGraphAsymmErrors*) f_elSF_looseToTight_endcap_17->Get("ratio");
-TH2F* h_elSF_recoToLoose_lowPt_17                = (TH2F*) f_elSF_recoToLoose_lowPt_17 ->Get("EGamma_SF2D");
-TH2F* h_elSF_recoToLoose_highPt_17               = (TH2F*) f_elSF_recoToLoose_highPt_17->Get("EGamma_SF2D");
-
-int getBinElectronLoose_17(float pt){
-	if     (pt >  5.0 && pt <= 12.5) return 0;
-	else if(pt > 12.5 && pt <= 16.0) return 1;
-	else if(pt > 16.0 && pt <= 20.0) return 2;
-	else if(pt > 20.0 && pt <= 25.0) return 3;
-	else if(pt > 25.0 && pt <= 30.0) return 4;
-	else if(pt > 30.0 && pt <= 40.0) return 5;
-	else if(pt > 40.0 && pt <= 75.0) return 6;
-	else if(pt > 75.0 && pt <= 120.0) return 7;
-	else {
-		assert(0);
-	}
-}
+	// Variations: conservative 5% also for the SF 
+	if(var>0) return SF*(1+0.05);
+	if(var<0) return SF*(1-0.05);
 
-float getElectronSFrecoToLoose_17(float _pt, float eta, int var = 0){
-	float pt = std::min(float(499.9), _pt); //protection
-	if(pt<20){
-		Int_t binx = (h_elSF_recoToLoose_lowPt_17->GetXaxis())->FindBin(eta);
-		Int_t biny = (h_elSF_recoToLoose_lowPt_17->GetYaxis())->FindBin(pt);
-		if(var>0) return (h_elSF_recoToLoose_lowPt_17->GetBinContent(binx, biny) + 0.01);
-		if(var<0) return (h_elSF_recoToLoose_lowPt_17->GetBinContent(binx, biny) - 0.01);
-		return  h_elSF_recoToLoose_lowPt_17->GetBinContent(binx, biny);
-	}
-	else {
-		Int_t binx = (h_elSF_recoToLoose_highPt_17->GetXaxis())->FindBin(eta);
-		Int_t biny = (h_elSF_recoToLoose_highPt_17->GetYaxis())->FindBin(pt);
-		if(var>0) return (h_elSF_recoToLoose_highPt_17->GetBinContent(binx, biny) + 0.01);
-		if(var<0) return (h_elSF_recoToLoose_highPt_17->GetBinContent(binx, biny) - 0.01);
-		return  h_elSF_recoToLoose_highPt_17->GetBinContent(binx, biny);
-	}
+	assert (SF>0 && "*** Warning we have a negative (or zero) Trigger SF ***");
+	return SF; 
 }
 
-float getElectronSFlooseToTight_17(float _pt, float eta, int var = 0){
-	// update this!
-	float pt = std::min(float(30.0), _pt); //protection
-	
-	if(abs(eta)<1.479){
-		if(var>0) return (h_elSF_looseToTight_barrel_17->Eval(pt) + h_elSF_looseToTight_barrel_17->GetErrorYhigh(getBinElectronLoose_17(pt))) ;
-		if(var<0) return (h_elSF_looseToTight_barrel_17->Eval(pt) - h_elSF_looseToTight_barrel_17->GetErrorYlow (getBinElectronLoose_17(pt))) ;
-		return  h_elSF_looseToTight_barrel_17->Eval(pt);
-	}
-	else {
-		if(var>0) return (h_elSF_looseToTight_endcap_17->Eval(pt) + h_elSF_looseToTight_endcap_17->GetErrorYhigh(getBinElectronLoose_17(pt))) ;
-		if(var<0) return (h_elSF_looseToTight_endcap_17->Eval(pt) - h_elSF_looseToTight_endcap_17->GetErrorYlow (getBinElectronLoose_17(pt))) ;
-		return h_elSF_looseToTight_endcap_17->Eval(pt);
-	}
+float triggerWZSF(float _pt1, float _eta1, int _pdgid1, float _pt2, float _eta2, int _pdgid2, float _pt3, float _eta3, int _pdgid3, float _met, float _met_corr, int _year, float var=0){
+	return 1.0;
```

Comment:
Dummy functions for the WZ trigger SF for now. It will be fixed in a following commit, when we know how to treat them.